src/crimson/os/seastore/async_cleaner.h

```diff
@@ -266,26 +266,44 @@ class segments_info_t {
   std::multiset<sea_time_point> modify_times;
 };
 
+struct randomblocks_info_t {
+public:
+  void update_latest_journal_head_committed(journal_seq_t seq) {
+    journal_head_committed = seq;
+  }
+  journal_seq_t get_journal_head() const {
+    return journal_head_committed;
+  }
+private:
+  journal_seq_t journal_head_committed;
+};
+
 /**
  * Callback interface for managing available segments
  */
-class SegmentProvider {
+class Provider {
```

Comment:
I agree with https://github.com/ceph/ceph/pull/46986#issuecomment-1183672601:

> I point this out simply because I'm not sure that it makes sense to couple journal trimming with cleaning -- they seem like distinct background operations.

In another word, it makes sense to decouple trimming out of **reclaimming**, but probably still keep the trimmer in the Cleaner as it is responsible to block incoming transactions.

The first step can be to split trim related interface out of SegmentProvider. In this case, rename `Provider` to `Trimmer` and making it an independent interface. What do you think?

Reply:
Agree.